test/retry-with-a-custom-handler.test.js

```diff
@@ -59,79 +59,79 @@ test('a 500 status code with no custom handler should fail', async (t) => {
+test('custom retry does not invoke the default delay causing a 501', async (t) => {
+  // the key here is our retryDelayHandler doesn't register the deefault handler and as a result it doesn't work
+  const customRetryLogic = ({ req, res, err, attempt, getDefaultDelay }) => {
     if (res && res.statusCode === 500 && req.method === 'GET') {
-      return 300
+      return 0
     }
     return null
   }
 
-  const { instance } = await setupServer(t, { customRetry: { handler: customRetryLogic, retries: 10 } }, 503)
+  const { instance } = await setupServer(t, { retryDelay: { handler: customRetryLogic } }, 501)
 
   let errorMessage
   try {
-    await got.get(`http://localhost:${instance.server.address().port}`, { retry: 0 })
+    await got.get(`http://localhost:${instance.server.address().port}`, { retry: 5 })
   } catch (error) {
     errorMessage = error.message
   }
 
-  t.equal(errorMessage, 'Response code 503 (Service Unavailable)')
+  t.equal(errorMessage, 'Response code 501 (Not Implemented)')
 })
 
 test('custom retry delay functions can invoke the default delay', async (t) => {
-  const customRetryLogic = ({ req, res, err, getDefaultDelay }) => {
+  const customRetryLogic = ({ req, res, err, attempt, getDefaultDelay }) => {
     // registering the default retry logic for non 500 errors if it occurs
     const defaultDelay = getDefaultDelay()
     if (defaultDelay) return defaultDelay
 
     if (res && res.statusCode === 500 && req.method === 'GET') {
-      return 300
+      return 0.1
     }
 
     return null
   }
 
-  const { instance } = await setupServer(t, { customRetry: { handler: customRetryLogic, retries: 10 } }, 503)
+  const { instance } = await setupServer(t, { customRetry: { handler: customRetryLogic } }, 500)
 
-  const res = await got.get(`http://localhost:${instance.server.address().port}`, { retry: 0 })
+  const res = await got.get(`http://localhost:${instance.server.address().port}`, { retry: 5 })
 
   t.equal(res.headers['content-type'], 'text/plain')
   t.equal(res.statusCode, 205)
   t.equal(res.body.toString(), 'Hello World 5!')
 })
 
 test('custom retry delay function inspects the err paramater', async (t) => {
-  const customRetryLogic = ({ req, res, err, getDefaultDelay }) => {
+  const customRetryLogic = ({ req, res, err, attempt, getDefaultDelay }) => {
     if (err && (err.code === 'UND_ERR_SOCKET' || err.code === 'ECONNRESET')) {
-      return 300
+      return 0.1
     }
     return null
   }
 
-  const { instance } = await setupServer(t, { customRetry: { handler: customRetryLogic, retries: 10 } }, 500, 4, true)
+  const { instance } = await setupServer(t, { retryDelay: { handler: customRetryLogic } }, 500, 4, true)
 
-  const res = await got.get(`http://localhost:${instance.server.address().port}`, { retry: 0 })
+  const res = await got.get(`http://localhost:${instance.server.address().port}`, { retry: 5 })
 
   t.equal(res.headers['content-type'], 'text/plain')
   t.equal(res.statusCode, 205)
```

Comment:
I think a test that asserts on the `attempt` property and notes it goes up would be good, happy to pair on it if you like